File: src/Find the Peak Element in a 2D Matrix/Code/solution.cpp
class Solution
{
public:
    vector<int> findPeakGrid(vector<vector<int>> &mat)
    {
        int n = mat.size();
        int m = mat[0].size();

        // Traverse every cell in the matrix
        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < m; j++)
            {

                int curr = mat[i][j];

                // Safely get neighbor values
                int up = (i > 0) ? mat[i - 1][j] : INT_MIN;
                int down = (i < n - 1) ? mat[i + 1][j] : INT_MIN;
                int left = (j > 0) ? mat[i][j - 1] : INT_MIN;
                int right = (j < m - 1) ? mat[i][j + 1] : INT_MIN;

                // Check peak condition
                if (curr >= up && curr >= down &&
                    curr >= left && curr >= right)
                {
                    return {i, j};
                }
            }
        }

        return {-1, -1};
    }
};

File: src/Find the Peak Element in a 2D Matrix/Code/solution.java
class Solution {
    public ArrayList<Integer> findPeakGrid(int[][] mat) {
        int n = mat.length;
        int m = mat[0].length;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {

                int curr = mat[i][j];

                int up = (i > 0) ? mat[i - 1][j] : Integer.MIN_VALUE;
                int down = (i < n - 1) ? mat[i + 1][j] : Integer.MIN_VALUE;
                int left = (j > 0) ? mat[i][j - 1] : Integer.MIN_VALUE;
                int right = (j < m - 1) ? mat[i][j + 1] : Integer.MIN_VALUE;

                if (curr >= up && curr >= down &&
                        curr >= left && curr >= right) {

                    ArrayList<Integer> ans = new ArrayList<>();
                    ans.add(i);
                    ans.add(j);
                    return ans;
                }
            }
        }
        return new ArrayList<>();
    }
}
